### PR TITLE
Added JSONMerge functions for MySQL

### PR DESCRIPTION
**Pull request by maikroeder**
Summary: *please summarize what this PR fixes/adds*

Checklist:

- [ ] Docs updated, or N/A
- [ ] Line added to HISTORY.rst, or N/A
- [ ] Added extra items to checklist as applicable
- [ ] All commits squashed into one.

Test Plan: <summarize how you tested your fix/addition here>

### CODE REVIEW

**Comment by maikroeder** [comment]
@normos

File: HISTORY.rst [hunk 1]
```diff
@@ -8,6 +8,9 @@ Pending
 
 .. Insert new release notes below this line
 
+* Added ``JSONMergePatch``, ``JSONMergePreserve`` database functions that
```
Comment (adamchainz) [inline HISTORY.rst]: needs a rebase

File: tests/testapp/test_functions.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
     ColumnGet, ConcatWS, Field, Floor, Greatest, If, JSONExtract, JSONInsert,
     JSONKeys, JSONLength, JSONReplace, JSONSet, LastInsertId, Least,
     RegexpInstr, RegexpReplace, RegexpSubstr, Round, Sign, UpdateXML,
-    XMLExtractValue,
+    XMLExtractValue, JSONMergePatch, JSONMergePreserve
```
Comment (adamchainz) [inline tests/testapp/test_functions.py]: isort will want this alphabetically sorted

File: django_mysql/models/functions.py [hunk 3]
```diff
@@ -323,6 +323,23 @@ class JSONReplace(BaseJSONModifyFunc):
 class JSONSet(BaseJSONModifyFunc):
     function = 'JSON_SET'
 
+# MySQL Merge Functions
```
Comment (adamchainz) [inline django_mysql/models/functions.py]: I don't think we need the comment

**Comment by adamchainz** [comment]
@maikroeder I know it's been a bit of time since Djangocon EU, hope you're doing well. Are you intending to finish this, or should I look into doing it?

**Comment by adamchainz** [comment]
I'm closing due to age, and the forthcoming universal JSONField in Django 3.1. I will be deprecating the JSONField from Django-MySQL in due course. It might be useful to implement the merge function in Django core.

**Pull request closed without merging**